script/dryrun

```diff
@@ -90,20 +97,32 @@ USE_DERIVED_DATASETS = {
 }
 
 
-def worker_entrypoint(sqlfile):
+
+
+def worker_entrypoint(public_project, sqlfile):
     sql = open(sqlfile).read()
     if sqlfile in USE_DERIVED_DATASETS:
         url = DERIVED_DATASETS_DRY_RUN_URL
     else:
         url = DRY_RUN_URL
+
     try:
+        request_data = {"dataset": basename(dirname(dirname(sqlfile))), "query": sql}
+
+        try:
+            # specify project ID for queries in public project
+            metadata = Metadata.of_sql_file(sqlfile)
+            if metadata.is_public_bigquery():
+                request_data["project"] = public_project
```

Comment:
This option is currently ignored by the Cloud Function.

Reply:
I think the question here is which project will Airflow be running the query in. I was assuming that Airflow would be running the query jobs in the `shared-prod` project, with the destination table being in the public project. If that's the case, then I'd expect we can dry run the query using the existing cloud function in shared-prod.

Reply:
True. 

I think right now the Cloud Function just dry runs the query file in the derived dataset without having an option to specify a different destination table.

Reply:
Oh wait, for dryrun that should not matter

Reply:
Yeah, for dry run, there is no destination table. So it's a matter of whether the cloud function has access to the referenced tables, and that the project the query is running in matches the project assumed by the table references in the query.